runners/spark/src/main/java/org/apache/beam/runners/spark/TestSparkRunner.java

```diff
@@ -133,22 +137,36 @@ public SparkPipelineResult run(Pipeline pipeline) {
             finishState,
             isOneOf(PipelineResult.State.STOPPED, PipelineResult.State.DONE));
 
-        // validate assertion succeeded (at least once).
-        int successAssertions = result.getAggregatorValue(PAssert.SUCCESS_COUNTER, Integer.class);
+        MetricQueryResults queryResults = result.metrics().queryMetrics(
+            MetricsFilter.builder().addNameFilter(
+                MetricNameFilter.named(PAssert.class, PAssert.SUCCESS_COUNTER)).build());
+        long successAssertions = 0;
+        for (MetricResult<Long> res : queryResults.counters()) {
+          successAssertions += res.attempted().longValue();
+        }
+
         Integer expectedAssertions = testSparkPipelineOptions.getExpectedAssertions() != null
             ? testSparkPipelineOptions.getExpectedAssertions() : expectedNumberOfAssertions;
         assertThat(
             String.format(
                 "Expected %d successful assertions, but found %d.",
                 expectedAssertions, successAssertions),
             successAssertions,
-            is(expectedAssertions));
+            is(expectedAssertions.longValue()));
+
+
         // validate assertion didn't fail.
-        int failedAssertions = result.getAggregatorValue(PAssert.FAILURE_COUNTER, Integer.class);
+        queryResults = result.metrics().queryMetrics(
+            MetricsFilter.builder().addNameFilter(
+                MetricNameFilter.named(PAssert.class, PAssert.FAILURE_COUNTER)).build());
+        long failedAssertions = 0;
+        for (MetricResult<Long> res : queryResults.counters()) {
+          failedAssertions += res.attempted().longValue();
```

Comment:
(This one should remain `attempted` - we should never commit the failure, so it should be impossible for the failure counter to ever go above zero in `committed` - but this does generically mean that we can't be guaranteed to have a completely accurate signal of failed assertions.)